File: libs/storage/src/main/java/smecalculus/bezmen/storage/SepulkaStateMapper.java
package smecalculus.bezmen.storage;

import org.mapstruct.Mapper;
import smecalculus.bezmen.core.SepulkaStateDm;

@Mapper
public interface SepulkaStateMapper {
    SepulkaStateEm.AggregateRoot toEdge(SepulkaStateDm.AggregateRoot state);

    SepulkaStateDm.AggregateRoot toDomain(SepulkaStateEm.AggregateRoot state);

    SepulkaStateEm.Touch toEdge(SepulkaStateDm.Touch state);

    SepulkaStateDm.Existence toDomain(SepulkaStateEm.Existence state);

    SepulkaStateEm.Preview toEdge(SepulkaStateDm.Preview state);

    SepulkaStateDm.Preview toDomain(SepulkaStateEm.Preview state);
}

File: apps/sepuling-kotlin/src/main/kotlin/smecalculus/bezmen/storage/SepulkaStateMapper.kt
package smecalculus.bezmen.storage

import io.mcarle.konvert.api.Konverter
import smecalculus.bezmen.core.SepulkaStateDm
import smecalculus.bezmen.mapping.EdgeMapper

@Konverter
interface SepulkaStateMapper : EdgeMapper {
    fun toEdge(state: SepulkaStateDm.AggregateRoot): SepulkaStateEm.AggregateRoot

    fun toDomain(state: SepulkaStateEm.AggregateRoot): SepulkaStateDm.AggregateRoot

    fun toEdge(state: SepulkaStateDm.Touch): SepulkaStateEm.Touch

    fun toDomain(state: SepulkaStateEm.Existence): SepulkaStateDm.Existence

    fun toDomain(state: SepulkaStateEm.Preview): SepulkaStateDm.Preview
}
